upstream-sha256: bf41088942d96b5c1abc8d1468aedfd7ec1f9183131944da490cef707f2686ee
actual-sha256: 7cf1edb7ff7d701b7d78dedea149acc0fe3afb76251944e1b6c928f981a03436
--- upstream.expected
+++ tolk.actual
@@ -11,7 +11,7 @@
         float val2 = (*(data2_256+(alu1+8)));
         float val3 = (*(data2_256+alu1));
         float4 val4 = (*((float4*)((data1_128+((Ridx0<<2)+(Lidx1<<6))))));
-        *(buf0+0) = ((*(buf0+0))+(val4[0]*val3)+(val4[1]*val1)+(val4[2]*val2)+(val4[3]*val0));
+        *(buf0+0) = ((*(buf0+0))+((val4[0]*val3)+(val4[1]*val1)+(val4[2]*val2)+(val4[3]*val0)));
       }
       float val5 = (*(data3_4+Lidx2));
       *(data0_8+((Lidx1<<2)+Lidx2)) = ((*(buf0+0))+val5);
@@ -34,19 +34,25 @@
   float4 val1 = (*((float4*)((data1_8+4))));
   float alu0 = (((val0[0]!=val0[0])^1)?val0[0]:((float)(-__builtin_inff())));
   float alu1 = (((val0[1]!=val0[1])^1)?val0[1]:((float)(-__builtin_inff())));
-  float alu2 = ((alu0<alu1)?alu1:alu0);
-  float alu3 = (((val0[2]!=val0[2])^1)?val0[2]:((float)(-__builtin_inff())));
-  float alu4 = ((alu2<alu3)?alu3:alu2);
-  float alu5 = (((val0[3]!=val0[3])^1)?val0[3]:((float)(-__builtin_inff())));
-  float alu6 = (((val1[0]!=val1[0])^1)?val1[0]:((float)(-__builtin_inff())));
-  float alu7 = (((val1[1]!=val1[1])^1)?val1[1]:((float)(-__builtin_inff())));
-  float alu8 = ((alu6<alu7)?alu7:alu6);
-  float alu9 = (((val1[2]!=val1[2])^1)?val1[2]:((float)(-__builtin_inff())));
-  float alu10 = ((alu8<alu9)?alu9:alu8);
-  float alu11 = (((val1[3]!=val1[3])^1)?val1[3]:((float)(-__builtin_inff())));
-  float alu12 = ((alu4<alu5)?alu5:alu4);
-  float alu13 = ((alu10<alu11)?alu11:alu10);
-  *((float2*)((data0_2+0))) = (float2){alu12,alu13};
+  float alu2 = ((alu0!=alu0)?alu0:alu1);
+  float alu3 = ((alu1<alu0)?alu0:alu2);
+  float alu4 = (((val0[2]!=val0[2])^1)?val0[2]:((float)(-__builtin_inff())));
+  float alu5 = ((alu3!=alu3)?alu3:alu4);
+  float alu6 = ((alu4<alu3)?alu3:alu5);
+  float alu7 = (((val0[3]!=val0[3])^1)?val0[3]:((float)(-__builtin_inff())));
+  float alu8 = (((val1[0]!=val1[0])^1)?val1[0]:((float)(-__builtin_inff())));
+  float alu9 = (((val1[1]!=val1[1])^1)?val1[1]:((float)(-__builtin_inff())));
+  float alu10 = ((alu8!=alu8)?alu8:alu9);
+  float alu11 = ((alu9<alu8)?alu8:alu10);
+  float alu12 = (((val1[2]!=val1[2])^1)?val1[2]:((float)(-__builtin_inff())));
+  float alu13 = ((alu11!=alu11)?alu11:alu12);
+  float alu14 = ((alu12<alu11)?alu11:alu13);
+  float alu15 = (((val1[3]!=val1[3])^1)?val1[3]:((float)(-__builtin_inff())));
+  float alu16 = ((alu6!=alu6)?alu6:alu7);
+  float alu17 = ((alu7<alu6)?alu6:alu16);
+  float alu18 = ((alu14!=alu14)?alu14:alu15);
+  float alu19 = ((alu15<alu14)?alu14:alu18);
+  *((float2*)((data0_2+0))) = (float2){alu17,alu19};
 }
 ---
 typedef float float4 __attribute__((aligned(16),ext_vector_type(4)));
@@ -140,19 +146,25 @@
   float4 val3 = (*((float4*)((data3_8+4))));
   float alu0 = (((val0[0]!=val1[0])&((val2[0]!=val2[0])^1))?val2[0]:((float)(-__builtin_inff())));
   float alu1 = (((val0[1]!=val1[0])&((val2[1]!=val2[1])^1))?val2[1]:((float)(-__builtin_inff())));
-  float alu2 = ((alu0<alu1)?alu1:alu0);
-  float alu3 = (((val0[2]!=val1[0])&((val2[2]!=val2[2])^1))?val2[2]:((float)(-__builtin_inff())));
-  float alu4 = ((alu2<alu3)?alu3:alu2);
-  float alu5 = (((val0[3]!=val1[0])&((val2[3]!=val2[3])^1))?val2[3]:((float)(-__builtin_inff())));
-  float alu6 = (((val0[0]!=val1[1])&((val3[0]!=val3[0])^1))?val3[0]:((float)(-__builtin_inff())));
-  float alu7 = (((val0[1]!=val1[1])&((val3[1]!=val3[1])^1))?val3[1]:((float)(-__builtin_inff())));
-  float alu8 = ((alu6<alu7)?alu7:alu6);
+  float alu2 = (((val0[0]!=val1[1])&((val3[0]!=val3[0])^1))?val3[0]:((float)(-__builtin_inff())));
+  float alu3 = (((val0[1]!=val1[1])&((val3[1]!=val3[1])^1))?val3[1]:((float)(-__builtin_inff())));
+  float alu4 = ((alu0!=alu0)?alu0:alu1);
+  float alu5 = ((alu1<alu0)?alu0:alu4);
+  float alu6 = (((val0[2]!=val1[0])&((val2[2]!=val2[2])^1))?val2[2]:((float)(-__builtin_inff())));
+  float alu7 = ((alu2!=alu2)?alu2:alu3);
+  float alu8 = ((alu3<alu2)?alu2:alu7);
   float alu9 = (((val0[2]!=val1[1])&((val3[2]!=val3[2])^1))?val3[2]:((float)(-__builtin_inff())));
-  float alu10 = ((alu8<alu9)?alu9:alu8);
-  float alu11 = (((val0[3]!=val1[1])&((val3[3]!=val3[3])^1))?val3[3]:((float)(-__builtin_inff())));
-  float alu12 = ((alu4<alu5)?alu5:alu4);
-  float alu13 = ((alu10<alu11)?alu11:alu10);
-  *((float2*)((data0_2+0))) = (float2){alu12,alu13};
+  float alu10 = ((alu5!=alu5)?alu5:alu6);
+  float alu11 = ((alu6<alu5)?alu5:alu10);
+  float alu12 = (((val0[3]!=val1[0])&((val2[3]!=val2[3])^1))?val2[3]:((float)(-__builtin_inff())));
+  float alu13 = ((alu8!=alu8)?alu8:alu9);
+  float alu14 = ((alu9<alu8)?alu8:alu13);
+  float alu15 = (((val0[3]!=val1[1])&((val3[3]!=val3[3])^1))?val3[3]:((float)(-__builtin_inff())));
+  float alu16 = ((alu11!=alu11)?alu11:alu12);
+  float alu17 = ((alu12<alu11)?alu11:alu16);
+  float alu18 = ((alu14!=alu14)?alu14:alu15);
+  float alu19 = ((alu15<alu14)?alu14:alu18);
+  *((float2*)((data0_2+0))) = (float2){alu17,alu19};
 }
 ---
 typedef int int2 __attribute__((aligned(8),ext_vector_type(2)));
@@ -277,10 +289,10 @@
     float alu19 = ((cast7<7.0f)?(cast7+-2.0f):6.0f);
     float alu20 = ((cast7<5.0f)?(cast7*0.5f):alu19);
     int alu21 = (Lidx3<<1);
-    *((float2*)((data0_16384+(alu21+12288)))) = (float2){(alu8*(1.0f+(((float)(((val3&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val8),(alu10*(1.0f+(((float)((alu6>>((unsigned char)(3u)))))*-2.0f))*val8)};
-    *((float2*)((data0_16384+(alu21+4096)))) = (float2){(alu3*(1.0f+(((float)(((val2&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val6),(alu5*(1.0f+(((float)((alu1>>((unsigned char)(3u)))))*-2.0f))*val6)};
-    *((float2*)((data0_16384+(alu21+8192)))) = (float2){(alu18*(1.0f+(((float)(((val5&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val7),(alu20*(1.0f+(((float)((alu16>>((unsigned char)(3u)))))*-2.0f))*val7)};
-    *((float2*)((data0_16384+alu21))) = (float2){(alu13*(1.0f+(((float)(((val4&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val9),(alu15*(1.0f+(((float)((alu11>>((unsigned char)(3u)))))*-2.0f))*val9)};
+    *((float2*)((data0_16384+(alu21+12288)))) = (float2){(alu8*(1.0f-(((float)(((val3&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val8),(alu10*(1.0f-(((float)((alu6>>((unsigned char)(3u)))))*2.0f))*val8)};
+    *((float2*)((data0_16384+(alu21+4096)))) = (float2){(alu3*(1.0f-(((float)(((val2&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val6),(alu5*(1.0f-(((float)((alu1>>((unsigned char)(3u)))))*2.0f))*val6)};
+    *((float2*)((data0_16384+(alu21+8192)))) = (float2){(alu18*(1.0f-(((float)(((val5&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val7),(alu20*(1.0f-(((float)((alu16>>((unsigned char)(3u)))))*2.0f))*val7)};
+    *((float2*)((data0_16384+alu21))) = (float2){(alu13*(1.0f-(((float)(((val4&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val9),(alu15*(1.0f-(((float)((alu11>>((unsigned char)(3u)))))*2.0f))*val9)};
   }
 }
 ---
@@ -304,10 +316,10 @@
       float4 val5 = (*((float4*)((data2_16384+(alu5+4096)))));
       float4 val6 = (*((float4*)((data2_16384+(alu5+8192)))));
       float4 val7 = (*((float4*)((data2_16384+alu5))));
-      *(buf0+0) = ((*(buf0+0))+(val3[0]*val7[0])+(val3[1]*val7[1])+(val3[2]*val7[2])+(val3[3]*val7[3]));
-      *(buf0+1) = ((*(buf0+1))+(val2[0]*val6[0])+(val2[1]*val6[1])+(val2[2]*val6[2])+(val2[3]*val6[3]));
-      *(buf0+2) = ((*(buf0+2))+(val3[0]*val5[0])+(val3[1]*val5[1])+(val3[2]*val5[2])+(val3[3]*val5[3]));
-      *(buf0+3) = ((*(buf0+3))+(val2[0]*val4[0])+(val2[1]*val4[1])+(val2[2]*val4[2])+(val2[3]*val4[3]));
+      *(buf0+0) = ((*(buf0+0))+((val3[0]*val7[0])+(val3[1]*val7[1])+(val3[2]*val7[2])+(val3[3]*val7[3])));
+      *(buf0+1) = ((*(buf0+1))+((val2[0]*val6[0])+(val2[1]*val6[1])+(val2[2]*val6[2])+(val2[3]*val6[3])));
+      *(buf0+2) = ((*(buf0+2))+((val3[0]*val5[0])+(val3[1]*val5[1])+(val3[2]*val5[2])+(val3[3]*val5[3])));
+      *(buf0+3) = ((*(buf0+3))+((val2[0]*val4[0])+(val2[1]*val4[1])+(val2[2]*val4[2])+(val2[3]*val4[3])));
     }
     float val8 = (((-1<val0[0])&(val0[0]<4))?*(data5_256+((val0[0]<<6)+Lidx3)):0.0f);
     float val9 = (((-1<val0[1])&(val0[1]<4))?*(data5_256+((val0[1]<<6)+Lidx3)):0.0f);
@@ -328,73 +340,77 @@
     float4 val1 = (*((float4*)((data1_256+alu0))));
     float alu1 = -val0[0];
     float alu2 = ((alu1<-7.0f)?-7.0f:alu1);
-    float alu3 = (alu2*2.4554669857025146f);
-    _Bool alu4 = (alu3!=alu3);
-    float alu5 = ((alu3!=((float)(-__builtin_inff())))?alu3:0.0f);
-    float alu6 = (alu4?0.0f:alu5);
-    float alu7 = ((alu3!=((float)(__builtin_inff())))?alu6:0.0f);
-    float alu8 = ((alu7<0.0f)?-0.5f:0.5f);
-    int cast0 = ((int)((alu7+alu8)));
-    float alu9 = -val0[2];
-    float alu10 = ((alu9<-7.0f)?-7.0f:alu9);
-    float alu11 = (alu10*2.4554669857025146f);
-    _Bool alu12 = (alu11!=alu11);
-    float alu13 = ((alu11!=((float)(-__builtin_inff())))?alu11:0.0f);
-    float alu14 = (alu12?0.0f:alu13);
-    float alu15 = ((alu11!=((float)(__builtin_inff())))?alu14:0.0f);
-    float alu16 = ((alu15<0.0f)?-0.5f:0.5f);
-    int cast1 = ((int)((alu15+alu16)));
-    float alu17 = -val1[0];
-    float alu18 = ((alu17<-7.0f)?-7.0f:alu17);
-    float alu19 = (alu18*2.4554669857025146f);
-    _Bool alu20 = (alu19!=alu19);
-    float alu21 = ((alu19!=((float)(-__builtin_inff())))?alu19:0.0f);
-    float alu22 = (alu20?0.0f:alu21);
-    float alu23 = ((alu19!=((float)(__builtin_inff())))?alu22:0.0f);
-    float alu24 = ((alu23<0.0f)?-0.5f:0.5f);
-    int cast2 = ((int)((alu23+alu24)));
-    float alu25 = -val1[2];
-    float alu26 = ((alu25<-7.0f)?-7.0f:alu25);
-    float alu27 = (alu26*2.4554669857025146f);
-    _Bool alu28 = (alu27!=alu27);
-    float alu29 = ((alu27!=((float)(-__builtin_inff())))?alu27:0.0f);
-    float alu30 = (alu28?0.0f:alu29);
-    float alu31 = ((alu27!=((float)(__builtin_inff())))?alu30:0.0f);
-    float alu32 = ((alu31<0.0f)?-0.5f:0.5f);
-    int cast3 = ((int)((alu31+alu32)));
-    int alu33 = (cast0>>1);
-    int alu34 = (cast1>>1);
-    int alu35 = (cast2>>1);
-    int alu36 = (cast3>>1);
-    float alu37 = ((val0[1]<-7.0f)?-7.0f:val0[1]);
-    float alu38 = -alu37;
-    float alu39 = (alu7-((float)(cast0)));
-    float alu40 = (((alu3<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu39)+0.0013392627006396651f)*alu39)+0.009618384763598442f)*alu39)+0.055503472685813904f)*alu39)+0.24022644758224487f)*alu39)+0.6931471824645996f)*alu39)+1.0f)*__builtin_bit_cast(float, (int)(((alu33+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu33)+127)<<23)))));
-    float alu41 = ((alu3<-150.0f)?0.0f:alu40);
-    float alu42 = (alu4?((float)(__builtin_nanf(""))):alu41);
-    float alu43 = ((alu38<-7.0f)?-7.0f:alu38);
-    float alu44 = ((val0[3]<-7.0f)?-7.0f:val0[3]);
-    float alu45 = -alu44;
-    float alu46 = (alu15-((float)(cast1)));
-    float alu47 = (((alu11<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu46)+0.0013392627006396651f)*alu46)+0.009618384763598442f)*alu46)+0.055503472685813904f)*alu46)+0.24022644758224487f)*alu46)+0.6931471824645996f)*alu46)+1.0f)*__builtin_bit_cast(float, (int)(((alu34+127)<<23)))*__builtin_bit_cast(float, (int)((((cast1-alu34)+127)<<23)))));
-    float alu48 = ((alu11<-150.0f)?0.0f:alu47);
-    float alu49 = (alu12?((float)(__builtin_nanf(""))):alu48);
-    float alu50 = ((alu45<-7.0f)?-7.0f:alu45);
-    float alu51 = ((val1[1]<-7.0f)?-7.0f:val1[1]);
-    float alu52 = -alu51;
-    float alu53 = (alu23-((float)(cast2)));
-    float alu54 = (((alu19<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu53)+0.0013392627006396651f)*alu53)+0.009618384763598442f)*alu53)+0.055503472685813904f)*alu53)+0.24022644758224487f)*alu53)+0.6931471824645996f)*alu53)+1.0f)*__builtin_bit_cast(float, (int)(((alu35+127)<<23)))*__builtin_bit_cast(float, (int)((((cast2-alu35)+127)<<23)))));
-    float alu55 = ((alu19<-150.0f)?0.0f:alu54);
-    float alu56 = (alu20?((float)(__builtin_nanf(""))):alu55);
-    float alu57 = ((alu52<-7.0f)?-7.0f:alu52);
-    float alu58 = ((val1[3]<-7.0f)?-7.0f:val1[3]);
-    float alu59 = -alu58;
-    float alu60 = (alu31-((float)(cast3)));
-    float alu61 = (((alu27<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu60)+0.0013392627006396651f)*alu60)+0.009618384763598442f)*alu60)+0.055503472685813904f)*alu60)+0.24022644758224487f)*alu60)+0.6931471824645996f)*alu60)+1.0f)*__builtin_bit_cast(float, (int)(((alu36+127)<<23)))*__builtin_bit_cast(float, (int)((((cast3-alu36)+127)<<23)))));
-    float alu62 = ((alu27<-150.0f)?0.0f:alu61);
-    float alu63 = (alu28?((float)(__builtin_nanf(""))):alu62);
-    float alu64 = ((alu59<-7.0f)?-7.0f:alu59);
-    *((float4*)((data0_128+(Lidx0<<2)))) = (float4){-((alu18/(alu56+1.0f))*(1.0f-alu57)),-((alu26/(alu63+1.0f))*(1.0f-alu64)),-((alu2/(alu42+1.0f))*(1.0f-alu43)),-((alu10/(alu49+1.0f))*(1.0f-alu50))};
+    float alu3 = -alu2;
+    float alu4 = (alu3*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu5 = (alu4!=alu4);
+    float alu6 = ((alu4!=((float)(-__builtin_inff())))?alu4:0.0f);
+    float alu7 = (alu5?0.0f:alu6);
+    float alu8 = ((alu4!=((float)(__builtin_inff())))?alu7:0.0f);
+    float alu9 = ((alu8<0.0f)?-0.5f:0.5f);
+    int cast0 = ((int)((alu8+alu9)));
+    float alu10 = -val0[2];
+    float alu11 = ((alu10<-7.0f)?-7.0f:alu10);
+    float alu12 = -alu11;
+    float alu13 = (alu12*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu14 = (alu13!=alu13);
+    float alu15 = ((alu13!=((float)(-__builtin_inff())))?alu13:0.0f);
+    float alu16 = (alu14?0.0f:alu15);
+    float alu17 = ((alu13!=((float)(__builtin_inff())))?alu16:0.0f);
+    float alu18 = ((alu17<0.0f)?-0.5f:0.5f);
+    int cast1 = ((int)((alu17+alu18)));
+    float alu19 = -val1[0];
+    float alu20 = ((alu19<-7.0f)?-7.0f:alu19);
+    float alu21 = -alu20;
+    float alu22 = (alu21*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu23 = (alu22!=alu22);
+    float alu24 = ((alu22!=((float)(-__builtin_inff())))?alu22:0.0f);
+    float alu25 = (alu23?0.0f:alu24);
+    float alu26 = ((alu22!=((float)(__builtin_inff())))?alu25:0.0f);
+    float alu27 = ((alu26<0.0f)?-0.5f:0.5f);
+    int cast2 = ((int)((alu26+alu27)));
+    float alu28 = -val1[2];
+    float alu29 = ((alu28<-7.0f)?-7.0f:alu28);
+    float alu30 = -alu29;
+    float alu31 = (alu30*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu32 = (alu31!=alu31);
+    float alu33 = ((alu31!=((float)(-__builtin_inff())))?alu31:0.0f);
+    float alu34 = (alu32?0.0f:alu33);
+    float alu35 = ((alu31!=((float)(__builtin_inff())))?alu34:0.0f);
+    float alu36 = ((alu35<0.0f)?-0.5f:0.5f);
+    int cast3 = ((int)((alu35+alu36)));
+    int alu37 = (cast0>>1);
+    int alu38 = (cast1>>1);
+    int alu39 = (cast2>>1);
+    int alu40 = (cast3>>1);
+    float alu41 = ((val0[1]<-7.0f)?-7.0f:val0[1]);
+    float alu42 = -alu41;
+    float alu43 = ((val0[3]<-7.0f)?-7.0f:val0[3]);
+    float alu44 = -alu43;
+    float alu45 = ((val1[1]<-7.0f)?-7.0f:val1[1]);
+    float alu46 = -alu45;
+    float alu47 = ((val1[3]<-7.0f)?-7.0f:val1[3]);
+    float alu48 = -alu47;
+    float alu49 = (alu8-((float)(cast0)));
+    float alu50 = (alu17-((float)(cast1)));
+    float alu51 = (alu26-((float)(cast2)));
+    float alu52 = (alu35-((float)(cast3)));
+    float alu53 = (((alu4<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu49)+0.0013392627006396651f)*alu49)+0.009618384763598442f)*alu49)+0.055503472685813904f)*alu49)+0.24022644758224487f)*alu49)+0.6931471824645996f)*alu49)+1.0f)*__builtin_bit_cast(float, (int)(((alu37+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu37)+127)<<23)))));
+    float alu54 = ((alu4<-150.0f)?0.0f:alu53);
+    float alu55 = (alu5?((float)(__builtin_nanf(""))):alu54);
+    float alu56 = (((alu13<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu50)+0.0013392627006396651f)*alu50)+0.009618384763598442f)*alu50)+0.055503472685813904f)*alu50)+0.24022644758224487f)*alu50)+0.6931471824645996f)*alu50)+1.0f)*__builtin_bit_cast(float, (int)(((alu38+127)<<23)))*__builtin_bit_cast(float, (int)((((cast1-alu38)+127)<<23)))));
+    float alu57 = ((alu13<-150.0f)?0.0f:alu56);
+    float alu58 = (alu14?((float)(__builtin_nanf(""))):alu57);
+    float alu59 = (((alu22<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu51)+0.0013392627006396651f)*alu51)+0.009618384763598442f)*alu51)+0.055503472685813904f)*alu51)+0.24022644758224487f)*alu51)+0.6931471824645996f)*alu51)+1.0f)*__builtin_bit_cast(float, (int)(((alu39+127)<<23)))*__builtin_bit_cast(float, (int)((((cast2-alu39)+127)<<23)))));
+    float alu60 = ((alu22<-150.0f)?0.0f:alu59);
+    float alu61 = (alu23?((float)(__builtin_nanf(""))):alu60);
+    float alu62 = (((alu31<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu52)+0.0013392627006396651f)*alu52)+0.009618384763598442f)*alu52)+0.055503472685813904f)*alu52)+0.24022644758224487f)*alu52)+0.6931471824645996f)*alu52)+1.0f)*__builtin_bit_cast(float, (int)(((alu40+127)<<23)))*__builtin_bit_cast(float, (int)((((cast3-alu40)+127)<<23)))));
+    float alu63 = ((alu31<-150.0f)?0.0f:alu62);
+    float alu64 = (alu32?((float)(__builtin_nanf(""))):alu63);
+    float alu65 = ((alu42<-7.0f)?-7.0f:alu42);
+    float alu66 = ((alu44<-7.0f)?-7.0f:alu44);
+    float alu67 = ((alu46<-7.0f)?-7.0f:alu46);
+    float alu68 = ((alu48<-7.0f)?-7.0f:alu48);
+    *((float4*)((data0_128+(Lidx0<<2)))) = (float4){((alu21/(alu61+1.0f))*(1.0f-alu67)),((alu30/(alu64+1.0f))*(1.0f-alu68)),((alu3/(alu55+1.0f))*(1.0f-alu65)),((alu12/(alu58+1.0f))*(1.0f-alu66))};
   }
 }
 ---
@@ -473,10 +489,10 @@
       float alu19 = ((cast7<7.0f)?(cast7+-2.0f):6.0f);
       float alu20 = ((cast7<5.0f)?(cast7*0.5f):alu19);
       int alu21 = ((Lidx3<<5)+(Lidx4_0<<1));
-      *((float2*)((data0_8192+(alu21+2048)))) = (float2){(alu3*(1.0f+(((float)(((val6&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val4),(alu5*(1.0f+(((float)((alu1>>((unsigned char)(3u)))))*-2.0f))*val4)};
-      *((float2*)((data0_8192+(alu21+4096)))) = (float2){(alu18*(1.0f+(((float)(((val9&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val2),(alu20*(1.0f+(((float)((alu16>>((unsigned char)(3u)))))*-2.0f))*val2)};
-      *((float2*)((data0_8192+(alu21+6144)))) = (float2){(alu8*(1.0f+(((float)(((val7&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val3),(alu10*(1.0f+(((float)((alu6>>((unsigned char)(3u)))))*-2.0f))*val3)};
-      *((float2*)((data0_8192+alu21))) = (float2){(alu13*(1.0f+(((float)(((val8&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val5),(alu15*(1.0f+(((float)((alu11>>((unsigned char)(3u)))))*-2.0f))*val5)};
+      *((float2*)((data0_8192+(alu21+2048)))) = (float2){(alu3*(1.0f-(((float)(((val6&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val4),(alu5*(1.0f-(((float)((alu1>>((unsigned char)(3u)))))*2.0f))*val4)};
+      *((float2*)((data0_8192+(alu21+4096)))) = (float2){(alu18*(1.0f-(((float)(((val9&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val2),(alu20*(1.0f-(((float)((alu16>>((unsigned char)(3u)))))*2.0f))*val2)};
+      *((float2*)((data0_8192+(alu21+6144)))) = (float2){(alu8*(1.0f-(((float)(((val7&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val3),(alu10*(1.0f-(((float)((alu6>>((unsigned char)(3u)))))*2.0f))*val3)};
+      *((float2*)((data0_8192+alu21))) = (float2){(alu13*(1.0f-(((float)(((val8&((unsigned char)(15u)))>>((unsigned char)(3u)))))*2.0f))*val5),(alu15*(1.0f-(((float)((alu11>>((unsigned char)(3u)))))*2.0f))*val5)};
     }
   }
 }
@@ -497,8 +513,9 @@
 void r_2_2(float* restrict data0_2, float* restrict data1_4) {
   for (int Lidx1 = 0; Lidx1 < 2; Lidx1++) {
     float2 val0 = (*((float2*)((data1_4+(Lidx1<<1)))));
-    float alu0 = ((val0[0]<val0[1])?val0[1]:val0[0]);
-    *(data0_2+Lidx1) = alu0;
+    float alu0 = ((val0[0]!=val0[0])?val0[0]:val0[1]);
+    float alu1 = ((val0[1]<val0[0])?val0[0]:alu0);
+    *(data0_2+Lidx1) = alu1;
   }
 }
 ---
@@ -524,14 +541,14 @@
     int alu12 = (cast0>>1);
     int alu13 = (cast1>>1);
     float alu14 = (alu4-((float)(cast0)));
-    float alu15 = (alu10-((float)(cast1)));
-    float alu16 = (((alu0<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu14)+0.0013392627006396651f)*alu14)+0.009618384763598442f)*alu14)+0.055503472685813904f)*alu14)+0.24022644758224487f)*alu14)+0.6931471824645996f)*alu14)+1.0f)*__builtin_bit_cast(float, (int)(((alu12+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu12)+127)<<23)))));
-    float alu17 = ((alu0<-150.0f)?0.0f:alu16);
-    float alu18 = (((alu6<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu15)+0.0013392627006396651f)*alu15)+0.009618384763598442f)*alu15)+0.055503472685813904f)*alu15)+0.24022644758224487f)*alu15)+0.6931471824645996f)*alu15)+1.0f)*__builtin_bit_cast(float, (int)(((alu13+127)<<23)))*__builtin_bit_cast(float, (int)((((cast1-alu13)+127)<<23)))));
-    float alu19 = ((alu6<-150.0f)?0.0f:alu18);
-    float alu20 = (alu1?((float)(__builtin_nanf(""))):alu17);
-    float alu21 = (alu7?((float)(__builtin_nanf(""))):alu19);
-    *(data0_2+Lidx1) = (alu20+alu21);
+    float alu15 = (((alu0<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu14)+0.0013392627006396651f)*alu14)+0.009618384763598442f)*alu14)+0.055503472685813904f)*alu14)+0.24022644758224487f)*alu14)+0.6931471824645996f)*alu14)+1.0f)*__builtin_bit_cast(float, (int)(((alu12+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu12)+127)<<23)))));
+    float alu16 = ((alu0<-150.0f)?0.0f:alu15);
+    float alu17 = (alu1?((float)(__builtin_nanf(""))):alu16);
+    float alu18 = (alu10-((float)(cast1)));
+    float alu19 = (((alu6<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu18)+0.0013392627006396651f)*alu18)+0.009618384763598442f)*alu18)+0.055503472685813904f)*alu18)+0.24022644758224487f)*alu18)+0.6931471824645996f)*alu18)+1.0f)*__builtin_bit_cast(float, (int)(((alu13+127)<<23)))*__builtin_bit_cast(float, (int)((((cast1-alu13)+127)<<23)))));
+    float alu20 = ((alu6<-150.0f)?0.0f:alu19);
+    float alu21 = (alu7?((float)(__builtin_nanf(""))):alu20);
+    *(data0_2+Lidx1) = (0.0f+alu17+alu21);
   }
 }
 ---
@@ -617,7 +634,7 @@
     float val14 = (((-1<val0[1])&(val0[1]<4))?*(data5_256+((val0[1]<<6)+Lidx4)):0.0f);
     float val15 = (((-1<val1[0])&(val1[0]<4))?*(data5_256+((val1[0]<<6)+Lidx4)):0.0f);
     float val16 = (((-1<val1[1])&(val1[1]<4))?*(data5_256+((val1[1]<<6)+Lidx4)):0.0f);
-    *(data0_128+(Lidx4+64)) = (((((*(buf0+2))+val16)*alu42)+(((*(buf0+3))+val14)*alu43))/val4[1]);
-    *(data0_128+Lidx4) = (((((*(buf0+0))+val15)*alu40)+(((*(buf0+1))+val13)*alu41))/val4[0]);
-  }
-}
+    *(data0_128+(Lidx4+64)) = ((0.0f+(((*(buf0+2))+val16)*alu42)+(((*(buf0+3))+val14)*alu43))/val4[1]);
+    *(data0_128+Lidx4) = ((0.0f+(((*(buf0+0))+val15)*alu40)+(((*(buf0+1))+val13)*alu41))/val4[0]);
+  }
+}
